src/app/index.js

```diff
@@ -1,30 +1,41 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
 
 import { NavBar } from '../components'
 import { MoviesList, MoviesInsert, MoviesUpdate, ConfirmRegistration, ResetPassword, ConfirmNewPassword, UserLogin, UserRegistration, UserAccount, ChangePassword, UpdateAccount, UsersList} from '../pages'
+import { authenticationService } from '../services/authentication'
 
 import 'bootstrap/dist/css/bootstrap.min.css'
 
 function App() {
+    const [currentUser, setCurrentUser] = useState(null);
+
+    useEffect(() => {
+        authenticationService.observableCurrentUser.subscribe(user => setCurrentUser(user))
```

Comment:
Maybe you need to unsubscribe when component unmount?

Check method "unsubscribe"

Someting like this:

useEffect(() => {
       const sub =  authenticationService.observableCurrentUser.subscribe(user => setCurrentUser(user))
        return () => sub.unsubscribe()
 }